test/unit/colorspace_spec.js

```diff
@@ -0,0 +1,563 @@
+
+      expect(colorSpace.getRgb(new Float32Array([0.1, 0.2, 0.3, 1]),
+        0)).toEqual(new Uint8Array(
+        [31, 27, 20]));
+      expect(colorSpace.getOutputLength(4, 0)).toEqual(3);
+      expect(colorSpace.isPassthrough(8)).toBeFalsy();
+      expect(testDest).toEqual(expectedDest);
+    });
+    it('should handle the case when cs is an indirect object', function () {
+      let cs = new Ref(10, 0);
+      let xref = new XRefMock([{
+        ref: cs,
+        data: new Name('DeviceCMYK'),
+      }]);
+      let res = new Dict();
+
+      let colorSpace = ColorSpace.parse(cs, xref, res);
+
+      let testSrc = new Uint8Array([27, 125, 250, 128,
+        131, 139, 140, 45,
+        111, 25, 198, 78,
+        21, 147, 255, 69
+      ]);
+      let testDest = new Uint8Array(3 * 3 * 3);
+      let expectedDest = new Uint8Array(
+        [135, 80, 18,
+          135, 80, 18,
+          113, 102, 97,
+          135, 80, 18,
+          135, 80, 18,
+          113, 102, 97,
+          112, 143, 75,
+          112, 143, 75,
+          188, 98, 27
+        ]
+      );
+      colorSpace.fillRgb(testDest, 2, 2, 3, 3, 3, 8, testSrc, 0);
+
+      expect(colorSpace.getRgb(new Float32Array([0.1, 0.2, 0.3, 1]),
+        0)).toEqual(new Uint8Array(
+        [31, 27, 20]));
+      expect(colorSpace.getOutputLength(4, 1)).toEqual(4);
+      expect(colorSpace.isPassthrough(8)).toBeFalsy();
+      expect(testDest).toEqual(expectedDest);
+    });
+  });
+
+  describe('CalGrayCS', function () {
+    it('should handle the case when cs is an array', function () {
+      let params = new Dict();
+      params.set('WhitePoint', [1, 1, 1]);
+      params.set('BlackPoint', [0, 0, 0]);
+      params.set('Gamma', 2.0);
+
+      let cs = new Array(2);
+      cs[0] = new Name('CalGray');
+      cs[1] = params;
+      let xref = new XRefMock([{
+        ref: new Ref(10, 0),
+        data: new Dict(),
+      }]);
+      let res = new Dict();
+
+      let colorSpace = ColorSpace.parse(cs, xref, res);
+
+      let testSrc = new Uint8Array([27, 125, 250, 131]);
+      let testDest = new Uint8Array(4 * 4 * 3);
+      let expectedDest = new Uint8Array(
+        [25, 25, 25,
+          25, 25, 25,
+          143, 143, 143,
+          143, 143, 143,
+          25, 25, 25,
+          25, 25, 25,
+          143, 143, 143,
+          143, 143, 143,
+          251, 251, 251,
+          251, 251, 251,
+          148, 148, 148,
+          148, 148, 148,
+          251, 251, 251,
+          251, 251, 251,
+          148, 148, 148,
+          148, 148, 148
+        ]
+      );
+      colorSpace.fillRgb(testDest, 2, 2, 4, 4, 4, 8, testSrc, 0);
+
+      expect(colorSpace.getRgb(new Float32Array([1.0]), 0)).toEqual(
+        new Uint8Array(
+          [255, 255, 255]));
+      expect(colorSpace.getOutputLength(4, 0)).toEqual(12);
+      expect(colorSpace.isPassthrough(8)).toBeFalsy();
+      expect(testDest).toEqual(expectedDest);
+    });
+  });
+  describe('CalRGBCS', function () {
```

Comment:
*This applies to other places in the patch as well.*

Please make sure that there is a newline before each `describe` block for readability.